server/channels/jobs/base_workers_resumable_test.go

```diff
@@ -0,0 +1,243 @@
+
+		// Send the job
+		select {
+		case worker.JobChannel() <- *job:
+			t.Log("Job sent to worker")
+		case <-time.After(time.Second):
+			require.Fail(t, "Failed to send job to worker")
+		}
+
+		// Let the job run for a bit
+		time.Sleep(200 * time.Millisecond)
+
+		// Stop the worker (should trigger setPendingOnStop)
+		t.Log("Stopping worker...")
+		go worker.Stop()
+
+		// Wait for the job to be cancelled
+		wg.Wait()
+
+		// Give some time for the pending status to be set
+		time.Sleep(100 * time.Millisecond)
+
+		// Verify all mocks were called
+		mockStore.JobStore.AssertExpectations(t)
+		mockMetrics.AssertExpectations(t)
+	})
+
+	t.Run("should not set job to pending for non-resumable worker", func(t *testing.T) {
+		jobServer, mockStore, mockMetrics := makeJobServer(t)
+
+		job := &model.Job{
+			Id:   "test-non-resumable-job-1",
+			Type: "test_non_resumable",
+			Data: map[string]string{},
+		}
+
+		var wg sync.WaitGroup
+		wg.Add(1)
+
+		execute := func(rctx request.CTX, job *model.Job) error {
+			defer wg.Done()
+			logger := rctx.Logger()
+			logger.Info("Non-resumable job started")
+
+			// Check for cancellation
+			select {
+			case <-rctx.Context().Done():
+				logger.Info("Non-resumable job cancelled")
+				return rctx.Context().Err()
+			case <-time.After(500 * time.Millisecond):
+				logger.Info("Non-resumable job completed")
+			}
+
+			return nil
+		}
+
+		isEnabled := func(_ *model.Config) bool {
+			return true
+		}
+
+		// Mock expectations - note we don't expect SetJobPending to be called
+		mockStore.JobStore.On("UpdateStatusOptimistically", "test-non-resumable-job-1", model.JobStatusPending, model.JobStatusInProgress).Return(job, nil)
+		mockStore.JobStore.On("UpdateOptimistically", mock.AnythingOfType("*model.Job"), model.JobStatusInProgress).Return(true, nil).Maybe()
+		mockStore.JobStore.On("Update", mock.AnythingOfType("*model.Job")).Return(nil, errors.New("test error")).Maybe()
+		mockMetrics.On("IncrementJobActive", "test_non_resumable")
+		mockMetrics.On("DecrementJobActive", "test_non_resumable")
+
+		// Create a regular (non-resumable) worker
+		worker := NewSimpleWorker("test_non_resumable", jobServer, execute, isEnabled)
+
+		// Start the worker
+		go worker.Run()
+
+		// Give the worker time to start
+		time.Sleep(50 * time.Millisecond)
+
+		// Send the job
+		select {
+		case worker.JobChannel() <- *job:
+			t.Log("Job sent to worker")
+		case <-time.After(time.Second):
+			require.Fail(t, "Failed to send job to worker")
+		}
+
+		// Let the job run for a bit
+		time.Sleep(200 * time.Millisecond)
+
+		// Stop the worker
+		t.Log("Stopping worker...")
+		go worker.Stop()
+
+		// Wait for the job to finish
+		wg.Wait()
+
+		// Give some time to ensure no pending status is set
+		time.Sleep(100 * time.Millisecond)
+
+		// Verify SetJobPending was NOT called
+		mockStore.JobStore.AssertNotCalled(t, "UpdateOptimistically", mock.AnythingOfType("*model.Job"), model.JobStatusPending)
+	})
```

Comment:
interesting, didn't know about that one, or maybe I have used it and just forgot. nice.